### ISSUE TITLE
Bug(destinations-BigQuery): Batch write failures with `Error 413 (Request Entity Too Large)`

### ISSUE THREAD

**Issue by erezrokah** [cloudquery/cloudquery#6070]
Could be related to https://discord.com/channels/872925471417962546/1057191097648033833/1057356930928082984 (waiting for further logs).

I was able to reproduce this error on our `cq-playground` account with the following configuration (I kept only the tables that generate the errors):

```yaml
kind: source
spec:
  name: aws
  version: v9.0.1
  destinations: [bigquery]
  path: cloudquery/aws
  tables: [aws_docdb_cluster_parameter_groups, aws_ec2_instance_types, aws_rds_engine_versions]
---
kind: destination
spec:
  name: bigquery
  path: cloudquery/bigquery
  version: "v1.3.1"
  write_mode: "append"
  spec:
    project_id: <project-id>
    dataset_id: <datatest-id>
```

Also, at some point it seems CloudQuery is hanging, the source plugin reports the sync finished, but the CLI is stuck on `Syncing resources`

**Comment by hermanschaaf** [comment]
@erezrokah Hmm, interesting. We should be able to fix this by trying to ensure we don't send batches above the max size, but as a workaround, does it work if you set `batch_size` in the spec to something lower, like 1 or 10?

**Comment by erezrokah** [comment]
Thanks @hermanschaaf I'll try with a smaller batch. I think there's another issue here as it seems the CLI deadlocks (I'm looking into it)

**Comment by erezrokah** [comment]
OK so discovered something else, `batch_size` at the plugin spec doesn't work anymore and need to be set on the top level spec:
```yml
kind: destination
spec:
  name: bigquery
  path: cloudquery/bigquery
  version: "v1.3.1"
  write_mode: "append"
  batch_size: 1 # This works
  spec:
    project_id: cq-playground
    dataset_id: erez_test
    batch_size: 1 # This doesn't
```

I'll open a PR to update the docs, and remove the configuration from relevant plugins

**Comment by erezrokah** [comment]
PR(s) to fix batching configs:
https://github.com/cloudquery/cloudquery/pull/6092
https://github.com/cloudquery/cloudquery/pull/6091

**Comment by erezrokah** [comment]
@hermanschaaf looks like using 1000 as batch size (the default before we switched to managed batching via the SDK) works 🎉 

https://github.com/cloudquery/cloudquery/pull/6092 should resolve this issue

**Comment by hermanschaaf** [comment]
@erezrokah Nice! It depends on the data in the rows though, so we should still check the size of the batch we're sending and cap it to some maximum. This might be a good thing to expose at the SDK level as well, similar to batch_size.

**Comment by kumargautam2534** [comment]
hi @erezrokah i had a discussion with you on discord regarding this issue.
so seems like using a batch of 1000 is working fine if data is getting fetched from 2-3 aws accounts.

But if i am running with a org account which some 50+ child accounts its ending up with this error "failed to terminate source client:  source plugin process exited with status signal: killed 
Error: failed to sync source aws: failed to fetch resources from stream: rpc error: code = Unavailable desc = error reading from server: read unix @->/tmp/cq-aXMVCYNaSXkMIqZP.sock: read: connection reset by peer"  (it took around 50mins before this error showed up on console)

**Comment by erezrokah** [comment]
Hi @kumargautam2534 thanks for reporting back and confirming the fix.

I think the current error you're experiencing might be related to the memory consumption of the sync command.

There are a few way to work around it:
1. Reduce concurrency (this will slow down the sync): see https://www.cloudquery.io/docs/reference/source-spec#concurrency
2. Split the configuration into multiple ones and run CloudQuery in parallel. See docs [here](https://www.cloudquery.io/docs/advanced-topics/running-cloudquery-in-parallel) and [here](https://www.cloudquery.io/docs/advanced-topics/performance-tuning)
3. Use a larger (memory) machine

**Comment by kumargautam2534** [comment]
Yes that vm have limited memory of 4gb only and it was maxing out during sync time. Thanks for your help.

**Comment by erezrokah** [comment]
Hi @kumargautam2534, FYI we resolved the original issue with [`v2.0.0` of the BigQuery plugin](https://github.com/cloudquery/cloudquery/blob/76faa0650878be9f4f03df2be3c9b53022d5961d/plugins/destination/bigquery/CHANGELOG.md#200-2022-12-29). With that version you don't need to set the batch size to 1000, as that's already the default.
